ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala

```diff
@@ -174,16 +174,19 @@ object AuthorizationInterceptor {
 
   private[auth] val contextKeyClaimSet = Context.key[ClaimSet]("AuthServiceDecodedClaim")
 
-  def extractClaimSetFromContext(): Try[ClaimSet] = {
+  def extractClaimSetFromContext(): Either[StatusRuntimeException, ClaimSet] = {
     val claimSet = contextKeyClaimSet.get()
-    if (claimSet == null)
-      Failure(
-        new RuntimeException(
-          "Thread local context unexpectedly does not store authorization claims. Perhaps a Future was used in some intermediate computation and changed the executing thread?"
-        )
+    if (claimSet == null) {
+      val msg = "Thread local context unexpectedly does not store authorization claims. " +
+        "Perhaps a Future was used in some intermediate computation and changed the executing thread?"
+      Left(
+        LedgerApiErrors.InternalError
```

Comment:
It's not directly related to the fix, but it struck me as odd not using error codes here